core/inspector.rs

```diff
@@ -500,6 +759,51 @@ pub struct SessionContainer {
   established: FuturesUnordered<InspectorSessionPumpMessages>,
   next_local_id: i32,
   local: HashMap<i32, Rc<InspectorSession>>,
+
+  target_sessions: HashMap<String, Rc<TargetSession>>, // sessionId -> TargetSession
+  main_session_id: Option<i32>, // The first session that should receive Target events
+  next_worker_id: u32, // Sequential ID for worker display naming (1, 2, 3, ...)
+}
+
+struct MainWorkerChannels {
+  main_to_worker_tx: UnboundedSender<String>,
+  worker_to_main_rx: UnboundedReceiver<InspectorMsg>,
+}
+
+/// Represents a CDP Target session (e.g., a worker)
+struct TargetSession {
+  target_id: String,
+  session_id: String,
+  local_session_id: i32,
+  /// Sequential worker ID for display (1, 2, 3, ...) - independent from session IDs
+  worker_id: u32,
+  main_worker_channels: RefCell<Option<MainWorkerChannels>>,
+  url: String,
+  /// Track if we've already sent attachedToTarget for this session
+  attached: Cell<bool>,
+}
+
+impl TargetSession {
+  /// Get a display title for the worker using Node.js style naming
+  /// e.g., "worker [1]", "worker [2]"
+  fn title(&self) -> String {
+    format!("worker [{}]", self.worker_id)
+  }
+
+  /// Send a message to the worker (main → worker direction)
+  fn send_to_worker(&self, message: String) {
+    if let Some(channels) = self.main_worker_channels.borrow().as_ref() {
+      let _ = channels.main_to_worker_tx.unbounded_send(message);
+    }
+  }
+
+  /// Poll for messages from the worker (worker → main direction)
+  fn poll_from_worker(&self, cx: &mut Context) -> Poll<Option<InspectorMsg>> {
+    match self.main_worker_channels.borrow_mut().as_mut() {
+      Some(channels) => channels.worker_to_main_rx.poll_next_unpin(cx),
+      None => Poll::Pending,
+    }
```

Comment:
This is a bit fishy - are we guaranteed to be woken up again if the `None` branch gets hit? It appears the context is only passed if there are some channels.

Reply:
Good catch, This should never have been executed like that. I added additional checks before this function to make and remove this None handler.